packages/core/src/interpreter.ts

```diff
@@ -838,21 +460,20 @@ export class Interpreter<
   }
   public toJSON() {
     return {
-      id: this.name
+      id: this.id
     };
   }
 
-  public [symbolObservable](): InteropSubscribable<
-    State<TContext, TEvent, TResolvedTypesMeta>
-  > {
+  public [symbolObservable](): InteropSubscribable<SnapshotFrom<TBehavior>> {
     return this;
   }
 
-  public getSnapshot() {
+  public getSnapshot(): SnapshotFrom<TBehavior> {
+    const getter = this.behavior.getSnapshot ?? ((s) => s);
     if (this.status === InterpreterStatus.NotStarted) {
-      return this.initialState;
+      return getter(this.getInitialState());
     }
```

Comment:
This is definitely a problem - at least with the current implementation of behaviors. `getInitialState` should be **pure** but unfortunately, it isn't for most of the implemented behaviors. 

The `Actor` interface mentions a `start` method:
https://github.com/statelyai/xstate/blob/ec4a4c185bbb3ceccd8004d2930a83e0074e845e/packages/core/src/types.ts#L1883

However, I think this isn't used anywhere right now (I could have missed something though). I think that a proper model here would perform those side-effects in `start`, moving them out of `getInitialState`